shared/common-utils.ts

```diff
@@ -102,10 +102,110 @@ export function addDigitSeparator(n: string, digitSeparator: string, chunkSize:
     return splitIntoChunks(n, chunkSize).join(digitSeparator);
 }
 
-export function splitArguments(options = ''): string[] {
-    // escape hashes first, otherwise they're interpreted as comments
-    const escapedOptions = options.replace(/#/g, '\\#');
-    return quoteParse(escapedOptions)
-        .map((x: any) => (typeof x === 'string' ? x : (x.pattern as string)))
-        .filter(Boolean);
+class ArgSplitter {
```

Comment:
not very important but maybe we could call it something like ArgumentParser or something

I can also imagine myself changing this to a parser combinator or something in the future perhaps

Reply:
Will rename: and you'd be welcome to rephrase as a combinator etc, I was trying to make minimal(ish) changes from a mostly-known-good source (and add enough tests to feel happy about things!)